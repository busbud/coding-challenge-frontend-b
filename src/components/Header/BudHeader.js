import React from 'react';

export default function BudHeader(props){
    return(
        <div className="header">
            <h1>Welcome to Osheaga 2020 event!<br></br>Please, take a seat!</h1>
            <h4>We are really pleased to help you finding your best ticket for the Osheaga festival 2020. Please, keep safe and have fun!</h4>
        </div>
    )
}