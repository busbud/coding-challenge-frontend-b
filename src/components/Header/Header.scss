@import "../../vars";
@import "../../susy";

.header {
    background-color: $light-blue;
    padding: $section-padding;

    &-link {
        display: block;
        width: 50%;
        max-width: 304px;
        margin: auto;

        &__logo {
            width: 100%;
        }
    }

    &__welcome-message {
        font-size: 52px;
        line-height: 68px;
        text-transform: uppercase;
        text-align: center;
        margin-bottom: 20px;
    }

}