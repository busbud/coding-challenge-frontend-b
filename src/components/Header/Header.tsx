import React from 'react';
import styled from 'styled-components';
import { Container } from '@material-ui/core';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { changeLang, langSelector, Lang } from '../../store/lang';
import { AppState } from '../../store';
import Button from '../Button';

interface Props {
  onClickAction: (payload: Lang) => void;
  lang: Lang;
}

const StyledHeader = styled.header`
  align-items: center;
  background-color: ${props => props.theme.busbud.header.backgroundColor};
  color: ${props => props.theme.busbud.header.color};
  display: flex;
  padding: ${props => props.theme.busbud.header.padding};
`;

const StyledContainer = styled(Container)`
  align-items: center;
  display: flex;
  justify-content: space-between;
`;

const UnordoredList = styled.ul``;
const ListItem = styled.li`
  display: inline;
  margin-left: 10px;
`;

function Header({ onClickAction }: Props) {
  const handleChangeLang = (newLang: Lang) => {
    onClickAction(newLang);
  };
  return (
    <StyledHeader>
      <StyledContainer>
        <FormattedMessage id="title" />
        <UnordoredList>
          <ListItem>
            <Button onClickAction={() => handleChangeLang('en')}>
              <FormattedMessage id="english" />
            </Button>
          </ListItem>
          <ListItem>
            <Button onClickAction={() => handleChangeLang('fr')}>
              <FormattedMessage id="french" />
            </Button>
          </ListItem>
        </UnordoredList>
      </StyledContainer>
    </StyledHeader>
  );
}

const mapStateToProps = (state: AppState) => ({
  lang: langSelector(state)
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  onClickAction: (lang: Lang) => dispatch(changeLang(lang))
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
