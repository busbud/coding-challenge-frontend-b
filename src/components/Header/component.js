import React from 'react';
import { object } from 'prop-types';
import AppBar from 'material-ui/AppBar';
import Grid from 'material-ui/Grid';
import Toolbar from 'material-ui/Toolbar';
import Logo from '../Logo';
import LocalesSelector from '../../containers/LocalesSelector';

const Component = ({ classes }) => (
  <AppBar className={classes.root} position="sticky" color="secondary">
    <Toolbar className={classes.toolbar}>
      <Grid container justify="center" className={classes.root}>
        <Grid item md={8} xs={11} className={classes.item}>
          <Logo />
          <LocalesSelector />
        </Grid>
      </Grid>
    </Toolbar>
  </AppBar>
);

Component.propTypes = {
  classes: object,
};

export default Component;
