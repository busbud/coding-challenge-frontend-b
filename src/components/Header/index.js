import { withStyles } from 'material-ui/styles';
import styles from './styles';
import component from './component';

export const enhance = withStyles(styles);

export default enhance(component);
