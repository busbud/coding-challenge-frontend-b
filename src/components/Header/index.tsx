import * as React from 'react'
import LangSwitcher from '../../components/LangSwitcher'
import CurrencySwitcher from '../../components/CurrencySwitcher'

interface TopBarProps {
  onLangChange: (ev: React.MouseEvent<HTMLButtonElement>) => void
  onCurrencyChange: (ev: React.MouseEvent<HTMLButtonElement>) => void
}

export default class TopBar extends React.Component<TopBarProps, any> {
  constructor (props: TopBarProps) {
    super(props)
  }

  render () {
    return (
      <div className='header-top-bar'>
        <div className='container'>
          <div className='row align-items-center'>
            <div className='col-md-3 header-top-bar-logo'></div>
            <div className='col-md-9 header-top-bar-lang'>
              <LangSwitcher langChange={this.props.onLangChange}/>
              <CurrencySwitcher currencyChange={this.props.onCurrencyChange}/>
            </div>
          </div>
        </div>
      </div>
    )
  }
}
