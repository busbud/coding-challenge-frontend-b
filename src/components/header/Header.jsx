import React from "react";

export const Header = () => <div>header</div>;
