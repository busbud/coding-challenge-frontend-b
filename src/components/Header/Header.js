import React, { PropTypes } from 'react';
import LangTogglerContainer from '../LangToggler/LangTogglerContainer';
import CurrencyTogglerContainer from '../CurrencyToggler/CurrencyTogglerContainer';
import './Header.scss';

/**
 *  Component
 **/

class Header extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <header className="container header">
        <a className="header-link" href="http://www.osheaga.com/" alt="Busbud's website">
          <img className="header-link__logo" src="assets/images/osheaga.png" alt="Osheaga logo" />
        </a>
        <h1 className="header__welcome-message">{this.props.translations.welcomeText}</h1>
        <LangTogglerContainer />
        <CurrencyTogglerContainer />
      </header>
    );
  }
}

Header.propTypes = {
    translations: PropTypes.object.isRequired
};


export default Header;