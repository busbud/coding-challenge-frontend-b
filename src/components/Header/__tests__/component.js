import React from 'react';
import { shallow } from 'enzyme';
import Header from '../component';
import Grid from 'material-ui/Grid';
import Toolbar from 'material-ui/Toolbar';

const props = {
  classes: {
    root: 'fakeRootClass',
    toolbar: 'fakeToolbarClass',
    item: 'fakeItemClass',
  },
};

describe('component |Â Header | component', () => {
  describe('render', () => {
    it('should render without throwing and error', () => {
      // given
      const wrapper = shallow(<Header {...props} />);

      // then
      expect(wrapper).toBeDefined();
    });

    it('should match snapshot', () => {
      // given
      const wrapper = shallow(<Header {...props} />);

      // then
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('AppBar', () => {
    it('should have proper className props', () => {
      // given
      const wrapper = shallow(<Header {...props} />);
      const className = wrapper.props().className;

      // then
      expect(className).toBe(props.classes.root);
    });
  });

  describe('Toolbar', () => {
    it('should have proper className props', () => {
      // given
      const wrapper = shallow(<Header {...props} />);
      const className = wrapper.find(Toolbar).props().className;

      // then
      expect(className).toBe(props.classes.toolbar);
    });
  });

  describe('Grid', () => {
    describe('container', () => {
      it('should have proper className props', () => {
        // given
        const wrapper = shallow(<Header {...props} />);
        const className = wrapper.findWhere(node => node.type() === Grid && node.props().container === true).props()
          .className;

        // then
        expect(className).toBe(props.classes.root);
      });
    });

    describe('item', () => {
      it('should have proper className props', () => {
        // given
        const wrapper = shallow(<Header {...props} />);
        const className = wrapper.findWhere(node => node.type() === Grid && node.props().item === true).props()
          .className;

        // then
        expect(className).toBe(props.classes.item);
      });
    });
  });
});
