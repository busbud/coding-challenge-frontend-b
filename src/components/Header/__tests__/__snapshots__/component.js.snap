// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component |Â Header | component render should match snapshot 1`] = `
<WithStyles(AppBar)
  className="fakeRootClass"
  color="secondary"
  position="sticky"
>
  <WithStyles(Toolbar)
    className="fakeToolbarClass"
  >
    <WithStyles(GridWrapper)
      className="fakeRootClass"
      container={true}
      justify="center"
    >
      <WithStyles(GridWrapper)
        className="fakeItemClass"
        item={true}
        md={8}
        xs={11}
      >
        <InjectIntl(WithStyles(Component)) />
        <withStateHandlers(Connect(Component)) />
      </WithStyles(GridWrapper)>
    </WithStyles(GridWrapper)>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
